### PR TITLE
Feature/improve dup comms

### PR DESCRIPTION
**Pull request by shuds13**
Use context management to free libE communicators.

### CODE REVIEW

File: libensemble/libE.py [hunk 1]
```diff
@@ -204,31 +216,26 @@ def libE_mpi(sim_specs, gen_specs, exit_criteria,
...
+            libE_nodes = set(comm.allgather(local_host))
+            exctr.add_comm_info(libE_nodes=libE_nodes, serial_setup=is_master)
+
+        # Run manager or worker code, depending
+        if is_master:
+            return libE_mpi_manager(comm, sim_specs, gen_specs, exit_criteria,
+                                    persis_info, alloc_specs, libE_specs, H0)
+
         # Worker returns a subset of MPI output
         libE_mpi_worker(comm, sim_specs, gen_specs, libE_specs)
```
Comment (jmlarson1) [inline libensemble/libE.py]: Is `worker_main` not returning anything? Can the `persis_info` in the next line be updated from `worker_main`?

Comment (jlnav) [reply libensemble/libE.py]: ``worker_main`` in it's current form just initiates worker instances. ``persis_info`` is first communicated to a worker in a unit of work through the communicator, and back to the manager through the communicator as well, in a loop initialized within ``worker_main`` but without anything returned to it.

Comment (shuds13) [reply libensemble/libE.py]: Yes, the worker does not return anything. It is merely to satisfy the interface. Perhaps it should return an empty dictionary?

Comment (jmlarson1) [reply libensemble/libE.py]: Sounds good

**Pull request merged**